src/realm/object-store/impl/deep_change_checker.cpp

```diff
@@ -226,25 +289,201 @@ bool DeepChangeChecker::check_row(Table const& table, ObjKeyType key, size_t dep
+            auto list = object.get_list<Mixed>(column_key);
+            for (size_t i = 0; i < list.size(); i++) {
+                auto target_object = list.get_any(i);
+                check_mixed_object(target_object);
+            }
+        }
+        else {
+            auto list = object.get_linklist(column_key);
+            for (size_t i = 0; i < list.size(); i++) {
+                auto target_object = list.get(i);
+                find_changed_columns(changed_columns, key_path, depth + 1, *target_table, target_object.value);
+            }
+        }
+    }
+    else if (column_key.is_set()) {
+        if (column_type == col_type_Mixed) {
+            auto set = object.get_set<Mixed>(column_key);
+            for (size_t i = 0; i < set.size(); i++) {
+                auto target_object = set.get(i);
+                check_mixed_object(target_object);
+            }
+        }
+        else {
+            auto set = object.get_linkset(column_key);
+            for (size_t i = 0; i < set.size(); i++) {
+                auto target_object = set.get(i);
+                find_changed_columns(changed_columns, key_path, depth + 1, *target_table, target_object.value);
+            }
+        }
+    }
+    else if (column_key.is_dictionary()) {
+        if (column_type == col_type_Mixed) {
+            auto dictionary = object.get_dictionary(column_key);
+            for (size_t i = 0; i < dictionary.size(); i++) {
+                auto target_object = dictionary.get(dictionary.get_key(i));
+                check_mixed_object(target_object);
+            }
+        }
+        else {
+            auto dictionary = object.get_dictionary(column_key);
+            auto linked_dictionary = std::make_unique<DictionaryLinkValues>(dictionary);
+            for (size_t i = 0; i < linked_dictionary->size(); i++) {
+                auto target_object = linked_dictionary->get_key(i);
+                find_changed_columns(changed_columns, key_path, depth + 1, *target_table, target_object.value);
+            }
+        }
+    }
+    else if (column_type == col_type_Link) {
+        // A forward link will only have one target object.
+        auto target_object = object.get<ObjKey>(column_key);
+        find_changed_columns(changed_columns, key_path, depth + 1, *target_table, target_object.value);
+    }
+    else if (column_type == col_type_BackLink) {
+        // A backlink can have multiple origin objects. We need to iterate over all of them.
+        auto origin_table = table.get_opposite_table(column_key);
+        auto origin_column_key = table.get_opposite_column(column_key);
+        size_t backlink_count = object.get_backlink_count(*origin_table, origin_column_key);
+        for (size_t i = 0; i < backlink_count; i++) {
+            auto origin_object = object.get_backlink(*origin_table, origin_column_key, i);
+            find_changed_columns(changed_columns, key_path, depth + 1, *origin_table, origin_object.value);
+        }
+    }
```

Comment:
```suggestion
} else {
    REALM_UNREACHABLE(); // unhandled column type
}
```

Reply:
Thank you for the suggestion. I've applied it.

Reply:
Seems like Github didn't format it correctly though. :D This is fixed with e9f60dcf6.